labelme/widgets/canvas.py

```diff
@@ -732,6 +732,28 @@ def paintEvent(self, event):
             drawing_shape.addPoint(self.line[1])
             drawing_shape.fill = True
             drawing_shape.paint(p)
+        elif (
+            self.createMode == "ai_polygon"
+            and self.current is not None
+        ):
+            drawing_shape = self.current.copy()
+            drawing_shape.addPoint(self.line[1])
+            points = self._ai_callback(
```

Comment:
This code should not be executed when pressing the ctrl key

Reply:
**Describe**

A new pull addresses two issues in the AI-polygon feature:

1. A bug caused by function labelme.utils.distancetoline. #1164 

https://user-images.githubusercontent.com/49972828/234761611-42eef991-4732-47bc-b772-26fff35d246b.mp4

2. The addition of `Mouse Double Click` to end drawing AI-polygon, making it more user-friendly.

**Issue one**
The first issue was related to a bug caused by function `labelme.utils.distancetoline `. This bug resulted in an error in point-to-line distance calculation when two polygon vertices were the same. 

![20230427124746](https://user-images.githubusercontent.com/49972828/234762007-45e6d5c7-8f57-40c9-8d8b-09bad3a23171.png)

So the system always finds a non-none nearestEdge, users had to add a new vertex before editing labels. To address this, I've modified the code (in Line 94 of `labelme.utils.qt.py`) to return a point-to-point distance instead of 0 in this case.

**Issue Two**
The second issue was adding the functionality of `Mouse Double Click` to end drawing AI-polygons, which makes it more user-friendly as users need no longer press "Ctrl" and click mouse left keys at the same time.